config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/JmsDestinationEntityBuilder.java

```diff
@@ -0,0 +1,142 @@
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.createElementWithTextContent;
+
+@Singleton
+@SuppressWarnings("squid:S2068") // sonarcloud believes this is a hardcoded password
+public class JmsDestinationEntityBuilder implements EntityBuilder {
+    private static final String STORED_PASSWORD_REF_FORMAT = "${secpass.%s.plaintext}";
+    private static final Integer ORDER = 1500;
+    
+    private final IdGenerator idGenerator;
+    
+    @Inject
+    JmsDestinationEntityBuilder(IdGenerator idGenerator) {
+        this.idGenerator = idGenerator;
+    }
+    
+    @Override
+    public List<Entity> build(Bundle bundle, BundleType bundleType, Document document) {
+        switch (bundleType) {
+            case DEPLOYMENT:
+                return bundle.getJmsDestinations().entrySet().stream()
+                        .map(e -> EntityBuilderHelper.getEntityWithOnlyMapping(CASSANDRA_CONNECTION_TYPE, e.getKey(), idGenerator.generate()))
+                        .collect(Collectors.toList());
+            case ENVIRONMENT:
+                return bundle.getJmsDestinations().entrySet().stream().map(e ->
+                        buildEntity(bundle, e.getKey(), e.getValue(), document)
+                ).collect(Collectors.toList());
+            default:
+                throw new EntityBuilderException("Unknown bundle type: " + bundleType);
+        }
+    }
+
+    private Entity buildEntity(Bundle bundle, String name, JmsDestination jmsDestination, Document document) {
+        String id = idGenerator.generate();
+        
+        // Build JMS Destination element.
+        Element jmsDestinationDetailEle = createElementWithAttributesAndChildren(
+                document,
+                JMS_DESTINATION_DETAIL,
+                ImmutableMap.of(ATTRIBUTE_ID, id),
+                createElementWithTextContent(document, NAME, name),
+                createElementWithTextContent(document, JMS_DESTINATION_NAME, jmsDestination.getDestinationName()),
+                createElementWithTextContent(document, INBOUND, jmsDestination.isInbound()),
+                createElementWithTextContent(document, ENABLED, "TODO"),
+                createElementWithTextContent(document, TEMPLATE, jmsDestination.isTemplate())
+        );
+        
+        Map<String, Object> destinationDetailsProps = new HashMap<>();
+        destinationDetailsProps.put(DESTINATION_TYPE, jmsDestination.getDestinationType());
+        destinationDetailsProps.put(PROPERTY_USERNAME, jmsDestination.getDestinationUsername());
+        if (jmsDestination.getDestinationPasswordRef() != null) {
+            destinationDetailsProps.put(PROPERTY_PASSWORD, String.format(STORED_PASSWORD_REF_FORMAT, jmsDestination.getDestinationPasswordRef()));
+        } else {
+            destinationDetailsProps.put(PROPERTY_PASSWORD, jmsDestination.getDestinationPassword());
+        }
+        
+//        destinationDetailsProps.put("replyType", "");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWdWyJripqfxNGPa-4iT&open=AWdWyJripqfxNGPa-4iT)

[ISSUE_KEY]: AWdWyJripqfxNGPa-4iT